go/mysql/client.go

```diff
@@ -564,6 +578,18 @@ func (c *Conn) writeHandshakeResponse41(capabilities uint32, scrambledPassword [
 		length++
 	}
 
+	// If the server supports CapabilityClientConnAttr and there are attributes to be
+	// sent, then indicate
+	var attrLength int
+	if capabilities&CapabilityClientConnAttr != 0 && len(attributes) > 0 {
+		capabilityFlags |= CapabilityClientConnAttr
+		for key, value := range attributes {
+			// 1 byte for key len + key + 1 byte for val len + val
+			attrLength += 1 + len(key) + 1 + len(value)
```

Comment:
This also is not guaranteed afaik to be 1 byte for the length, these are length encoded strings according to the docs, so they should be parsed as such then.

Reply:
Yep. Agree in light of above.